### DISCUSSION TITLE
Deploying to Two Vercel Projects with a Shared GitHub Repository Across Different Branches

### DISCUSSION THREAD

**Discussion by chenhanting-ibuypower** [vercel/next.js#62175] [Help]
### Deploying to Two Vercel Projects with a Shared GitHub Repository Across Different Branches

I have two Vercel projects sharing the same repository in order to create two production environments. The token belongs to my account. How can I use the project_id provided by Vercel to deploy? 

```sh
vercel deploy --token gnZ2TRwYLa5pqTfe6U2J4tWE --name hyte-production-replica
Vercel CLI 29.2.1
❗️  The "--name" option is deprecated (https://vercel.link/name-flag)
🔍  Inspect: https://vercel.com/hyte/hyte/A4bR5PXwmG7MXNTiYQo8rmVCyr7K [8s]
⠴ Building
```

We have two projects: one is `hyte`, and the other is `hyte-production-replica`. `hyte-production-replica` is meant to create an environment similar to our live site. Previously, I wanted to specify the project by using the `--name` tag, but it didn't work.

I've checked this [link](https://vercel.link/name-flag), but it seems it doesn't apply to my use-case.

I have some workaround approaches, such as duplicating the repository or porting our production-replica environment over to Azure to create a completely isolated space. This ensures that when I execute the vercel deploy ... command, it won't interfere with each other.

It's still a pain for me because if I could stay completely on Vercel and use the most streamlined way, it would be great. Has anyone encountered a similar situation?

**Comment by vaidkaran** [comment]
This worked for me: https://github.com/vercel/vercel/discussions/4622#discussioncomment-7982941
